src/Backend/Modules/Pages/Domain/ModuleExtra/ModuleExtraRepository.php

```diff
@@ -51,4 +55,20 @@ public function getBlocks(): array
             ->getQuery()
             ->getResult();
     }
+
+    public function findWidgetsByModuleAndAction(string $module, string $action): array
```

Comment:
this isn't checking for widgets only. I would add the ModuleExtraType here as well as a parameter and just name it `findModuleExtra` or something and have module, action and type as parameters